### PR TITLE
[CCR] Add qa test library

### PR DESCRIPTION
**Pull request by martijnvg**
Introduced test qa lib that all CCR qa modules depend on in order to avoid test code duplication. 
All qa multi node tests now extend from a new base class that adds several helper methods.

### CODE REVIEW

File: x-pack/plugin/ccr/qa/chain/build.gradle [hunk 1]
```diff
@@ -5,6 +5,7 @@ apply plugin: 'elasticsearch.standalone-test'
 dependencies {
     testCompile project(path: xpackModule('core'), configuration: 'testArtifacts')
     testCompile project(path: xpackModule('ccr'), configuration: 'runtime')
+    testCompile project(':x-pack:plugin:ccr:qa:test-common')
```
Comment (nik9000) [inline x-pack/plugin/ccr/qa/chain/build.gradle]: SQL has this and it sticks it in `qa`.

Comment (nik9000) [reply x-pack/plugin/ccr/qa/chain/build.gradle]: Well, it is in `x-pack/qa/sql` now, but I'm in the process of moving it to `x-pack/plugin/sql/qa`.

Comment (martijnvg) [reply x-pack/plugin/ccr/qa/chain/build.gradle]: That is a much better idea than adding another gradle module. So I did this: 633afa77dd1cea40404d8f2de37a62674b39ec4a

Comment (martijnvg) [reply x-pack/plugin/ccr/qa/chain/build.gradle]: and here is the second half of the change: 772a4f85695310ad13f93a4283a0a08f9b281f6e

File: x-pack/plugin/ccr/qa/build.gradle [hunk 2]
```diff
@@ -1,5 +1,14 @@
 import org.elasticsearch.gradle.test.RestIntegTestTask
 
+apply plugin: 'elasticsearch.build'
+
+archivesBaseName = 'ccr-qa-test-common'
```
Comment (jasontedor) [inline x-pack/plugin/ccr/qa/build.gradle]: Let us remove this.

**Review by jasontedor — APPROVED** [review]
LGTM, left one comment, no need for another round.

**Pull request merged**